src/apps/store/services/promotion.service.ts

```diff
@@ -0,0 +1,57 @@
+import { PromotionRepository } from '@apps/store/repositories/promotion.repository';
+import { ResourceOutput, ResourceDataOutput } from '@shared/type';
+import { Service } from 'typedi';
+import { Errors } from '@apps/account/configs/errors';
+import { AppException } from '@shared/libs/exception';
+
+export type PromotionOutput = ResourceOutput & {
+  couponCode: string;
+  percentDiscount: number;
+  expirationDate: Date;
+};
+export type CreatePromotionInput = {
+  percentDiscount: number;
+  expirationDate: Date;
+};
+
+@Service()
+export class PromotionService {
+  constructor(private readonly promotionRepository: PromotionRepository) {}
+  async createGet(
+    data: CreatePromotionInput,
+  ): Promise<ResourceDataOutput<PromotionOutput>> {
+    const couponCode = this.generateRandomString(13);
+    const isExisted = await this.promotionRepository.findOne({
+      couponCode: couponCode,
+    });
+    if (isExisted) {
+      const { errorCode, message, httpCode } = Errors.COUPON_CODE_IS_EXISTED;
+      throw new AppException(errorCode, message, httpCode);
+    }
+    const entity = await this.promotionRepository.createGet({
+      ...data,
+      couponCode,
+    });
+    return {
+      data: {
+        id: entity.id,
+        couponCode: entity.couponCode,
+        percentDiscount: entity.percentDiscount,
+        expirationDate: entity.expirationDate,
+        createdAt: entity?.createdAt,
+        updatedAt: entity.updatedAt,
+      },
+    };
+  }
+  generateRandomString = (length) => {
```

Comment:
The method declaration should be consistent in the class.
`generateRandomString = (length) => {}` 
=> 
`generateRandomString(length) {}`